common/models/User-Identity.js

```diff
@@ -145,27 +147,42 @@ export default function(UserIdent) {
         user.picture = picture;
         userChanged = true;
       }
-
       if (!githubRegex.test(provider) && profile) {
         user[provider] = getUsernameFromProvider(provider, profile);
         userChanged = true;
+        if (userChanged) {
```

Comment:
I am out at the moment but I will fix everything that you mentioned later, thanks for the feedback